From e9c3cb7cc9077153abdbb656b2c411ade1c3a8b9 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:12:25 -0400
Subject: [powerpc] tm: Remove unnecessary r1 save

Message-id: <1410545655-205645-517-git-send-email-dzickus@redhat.com>
Patchwork-id: 94403
O-Subject: [RHEL7 PATCH 516/626] powerpc/tm: Remove unnecessary r1 save
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit ce0ac1fc326b6a4116728be933ff46d75269baa1
Author: Michael Neuling <mikey@neuling.org>
Date:   Fri Mar 28 16:40:33 2014 +1100

    powerpc/tm: Remove unnecessary r1 save

    We save r1 to the scratch SPR and restore it from there after the trechkpt so
    saving r1 to the paca is not needed.

    Signed-off-by: Michael Neuling <mikey@neuling.org>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kernel/tm.S b/arch/powerpc/kernel/tm.S
index ddf84c2..9631266 100644
--- a/arch/powerpc/kernel/tm.S
+++ b/arch/powerpc/kernel/tm.S
@@ -319,8 +319,6 @@ _GLOBAL(__tm_recheckpoint)
   */
  SAVE_NVGPRS(r1)
 
- std r1, PACAR1(r13)
-
  /* Load complete register state from ts_ckpt* registers */
 
  addi r7, r3, PT_CKPT_REGS  /* Thread's ckpt_regs */
-- 
1.7.1